### PR TITLE
 JDP230402-15 Added UserRepository, User, Cart and Order

### PR DESCRIPTION
**Pull request by KamilBonar**

### CODE REVIEW

**Review by michal2220 — APPROVED** [review]
Looks ok for me :)

**Review by JabAlex — APPROVED** [review]
Looks ok

File: src/main/java/com/kodilla/ecommercee/domain/Cart.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package com.kodilla.ecommercee.domain;
+
+import com.sun.istack.NotNull;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.*;
+
+@NoArgsConstructor
+@Data
```
Comment (WitoZak) [inline src/main/java/com/kodilla/ecommercee/domain/Cart.java]: Using '.Data' can lead to performance issues if there are many fields, especially if some of these fields are not used frequently. Use instead '@Getter' '@Setter'

File: src/main/java/com/kodilla/ecommercee/domain/Order.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package com.kodilla.ecommercee.domain;
+
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.jetbrains.annotations.NotNull;
+
+import javax.persistence.*;
+
+@NoArgsConstructor
+@Data
```
Comment (WitoZak) [inline src/main/java/com/kodilla/ecommercee/domain/Order.java]: Using '.Data' can lead to performance issues if there are many fields, especially if some of these fields are not used frequently. Use instead '@Getter' '@Setter'

File: src/main/java/com/kodilla/ecommercee/domain/User.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    @Column(name = "USERNAME")
+    private String username;
+
+    @NotNull
+    @Column(name = "STATUS")
+    private boolean status;
+
+    @NotNull
+    @Column(name = "USER_KEY")
+    private String userKey;
```
Comment (WitoZak) [inline src/main/java/com/kodilla/ecommercee/domain/User.java]: Don't You think that userKey should be int or Long ?

**Comment by WitoZak** [comment]
If possible, please change the name of the pull request and its description to match the task.

File: src/main/java/com/kodilla/ecommercee/domain/Cart.java (hunk 1 above)
Comment (KamilBonar) [reply src/main/java/com/kodilla/ecommercee/domain/Cart.java]: I changed it.

File: src/main/java/com/kodilla/ecommercee/domain/Order.java (hunk 2 above)
Comment (KamilBonar) [reply src/main/java/com/kodilla/ecommercee/domain/Order.java]: I changed it.

File: src/main/java/com/kodilla/ecommercee/domain/User.java (hunk 3 above)
Comment (KamilBonar) [reply src/main/java/com/kodilla/ecommercee/domain/User.java]: I changed to Long.

**Pull request merged**